lib/skate/detours/db/route_pattern.ex

```diff
@@ -0,0 +1,54 @@
+# Should this live here under detours?
+# I don't like how the db folders are scattered throughout /lib subdirectories.
+# I'd rather they all live in one subfolder. 
+# Example: there's db files in /notifications and /skate/detours and /skate/settings, etc
+defmodule Skate.Detours.Db.RoutePattern do
+  @moduledoc """
+  Ecto Model for `route_patterns` Database table
+  """
+
+  use Skate.Schema
+  import Ecto.Changeset
+
+  alias Skate.Detours.Db.Detour
+  alias Skate.Detours.DirectionName
+
+  @required_fields [
+    :hash,
+    :gtfs_route_pattern_id,
+    :gtfs_route_pattern_name,
+    :gtfs_route_pattern_headsign,
+    # :gtfs_route_pattern_shape,
+    :gtfs_route_pattern_direction_name,
+    :gtfs_route_id,
+    :gtfs_route_name
+  ]
+
+  typed_schema "route_patterns" do
+    field(:hash, :integer)
+    field(:gtfs_route_pattern_id, :string)
+    field(:gtfs_route_pattern_name, :string)
+    field(:gtfs_route_pattern_headsign, :string)
+    field(:gtfs_route_pattern_direction_name, DirectionName)
+
+    # The given route pattern will always have the given route
+    # Does it need to be stored here as well?
+    field(:gtfs_route_id, :string)
+    field(:gtfs_route_name, :string)
+
+    field(:gtfs_route_pattern_time_description, :string)
```

Comment:
I didn't know this existed! I think we should set [`writable: :insert`](https://hexdocs.pm/ecto/Ecto.Schema.html#field/3) on these fields so we don't accidentally update them

```suggestion
    field(:gtfs_route_pattern_id, :string, writable: :insert)
    field(:gtfs_route_pattern_name, :string, writable: :insert)
    field(:gtfs_route_pattern_headsign, :string, writable: :insert)
    field(:gtfs_route_pattern_direction_name, DirectionName, writable: :insert)

    # The given route pattern will always have the given route
    # Does it need to be stored here as well?
    field(:gtfs_route_id, :string, writable: :insert)
    field(:gtfs_route_name, :string, writable: :insert)

    field(:gtfs_route_pattern_time_description, :string)
```